### ISSUE TITLE
False positive with global variable

### ISSUE THREAD

**Issue by Keerecles** [google/sanitizers#1044]
Hi,
  In my program, the ASAN report **ERROR global-buffer-overflow**  on several global variables.However, 
I'm sure that I use the global variables in the right way, like this:
  In  AAA.c  
     //define a global var
       int g_var_a;
      ........
      ........
       g_var_a =1;  //ASAN report global-buffer-overflow right here!
  According to the formula, Shadow = (Mem >> 3) + 0x20000000,  the value of the shadow memory of the global var **g_var_a**  is F9 ( Global Redzone),witch  confuses me very much!
Is there any opinions?

**Comment by vitalybuka** [comment]
Do you compile all files with asan?

**Comment by Keerecles** [comment]
No. Only part of the files and all the dynamic libraries  were complied with Asan. In another project of mine, I compiled it in the same way and it worked well.    @vitalybuka

**Comment by vitalybuka** [comment]
Which version of compiler and runtime do you use?
also you can try set ASAN_OPTIONS=verbosity=1:report_globals=2 and see how asan register this variable. Maybe it will give some clues.

**Comment by Keerecles** [comment]
> Which version of compiler and runtime do you use?
I use GCC 8.1
> also you can try set ASAN_OPTIONS=verbosity=1:report_globals=2 and see how asan register this variable. Maybe it will give some clues.
Ok, I will try it, thank you!

**Comment by Keerecles** [comment]
Sadly, there is no useful imformation in the log when I set ASAN_OPTIONS=verbosity=1:report_globals=2.

**Comment by Keerecles** [comment]
> Does this happen with fresh clang?

No. I use GCC8.1 to compile my project on SUSE Linux Enterprise Server 12 
> Can you provide a minimized reproducer?

Sorry， my project is very complex and I can not provide many details.
The application is compiled  like this:

1.  A.c  +  B.c   --GCC8.1 ASan -->  libproblem.so
2.  C.c  +  libproblem.so  --GCC8.1 ASan --> App
3.  In A.c  :
        ... globals ...
        unsigned int g_var =0;
        ... globals ...

     In B.c :
        extern unsigned int g_var =0;
        ..........
        void fun(){
            .....
            g_var =0;       //ASan report global-buffer-overflow right here!
            .....
        }
      
 The type of the variable "g_var"  is **unsigned int**, not randomly picked.
